eng/packaging.targets

```diff
@@ -42,4 +51,63 @@
     <MSBuild Projects="$(PkgProjPath)"
              Targets="Build" />
   </Target>
+    
+  <Target Name="AddRefAssemblies" Condition="'$(ExcludeFromPackage)' == '' and '$(TargetsAnyOS)' == 'true'"> 
+    <PropertyGroup>
+      <RefPath>$(BaseOutputPath)/ref/$(TargetFrameworkWithoutSuffix)-$(Configuration)/$(TargetName).dll</RefPath>
+    </PropertyGroup>
+    
+    <ItemGroup Condition="Exists('$(RefPath)')">
```

Comment:
I don't like having "lossy" content.  IMHO we should have reference assembly inclusion be opt-in, and when it is we should ask the Reference project (via P2P) what it's package content should be.  The way this is structured now bad authoring or inconsistent ref/src will just silently drop some refs.  This may be irrelevant if we don't have any packages that *need* references.

Reply:
I see there's already a conversation going on here about the same thing: https://github.com/dotnet/runtime/pull/54250#discussion_r653136450